src/Common/Commands.Common.Test/TestClient.cs

```diff
@@ -0,0 +1,38 @@
+﻿/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using Microsoft.PowerBI.Api.V2;
+using Microsoft.PowerBI.Common.Abstractions.Interfaces;
+using Microsoft.PowerBI.Common.Client;
+using Microsoft.Rest;
+
+namespace Microsoft.PowerBI.Commands.Common.Test
+{
+    public class TestClient : IPowerBIClientFactory
+    {
+        public FakeHttpClientHandler ClientHandler { get; }
+
+        public TestClient(FakeHttpClientHandler handler)
+        {
+            this.ClientHandler = handler;
+        }
+
+        public IPowerBIClient CreateClient(IAuthenticationFactory authenticator, IPowerBIProfile profile, IPowerBILogger logger, IPowerBISettings settings)
+        {
+            var token = authenticator.Authenticate(profile, logger, settings);
+            if (Uri.TryCreate(profile.Environment.GlobalServiceEndpoint, UriKind.Absolute, out Uri baseUri))
+            {
+                return new PowerBIClient(baseUri, new TokenCredentials(token.AccessToken), this.ClientHandler);
```

Comment:
It looks like the intent behind this is to returned a version of the PowerBIClient that uses your fake HTTP responses. Why not simplify the flow and return a fake or mockable IPowerBIClient instead?

Reply:
I had issues getting the right input to trigger the result to come back. I think we should support both approaches for testing